src/cider/nrepl/middleware/util/instrument.clj

```diff
@@ -164,12 +164,16 @@
 (defn- contains-recur?
   "Return true if form is not a `loop` or a `fn` and a `recur` is found in it."
   [form]
-  (if (seq? form)
-    (case (first form)
-      recur true
-      loop* false
-      fn*   false
-      (some contains-recur? (rest form)))))
+  (cond (seq? form) (case (first form)
+                      recur true
+                      loop* false
+                      fn*   false
+                      (some contains-recur? (rest form)))
+        ;; case* expands the non-default branches into a map
+        ;; this depends on internal logic
+        (map? form) (some contains-recur? (vals form))
+        (vector? form) (some contains-recur? (seq form))
+        :else false))
```

Comment:
I don't know if this is the best way to handle the issue. I am open for other suggestions. I mean normally a `recur` should only appear in a map when used inside a `case`.

Reply:
I think your approach is pretty reasonable, but perhaps @Malabarba or @vspinu  have some better ideas.